### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by JeanMertz**
Initial Webmachine implementation

things still to be added in upcoming versions:

* Roar implementation
* Validators
* Better documentation
* More tests

### CODE REVIEW

File: features/webmachine.feature [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    When the client does a GET request to the "items" resource
+    Then the status code should be "200" (OK)
+    And the response should be HAL/JSON:
+      """json
+      {
+        "_links": {
+          "self": {
+            "href": "https://example.org/items"
+          },
+          "b:items": [
```
Comment (eterps) [inline features/webmachine.feature]: Maybe better to remove this namespace or add the curies attr?

File: features/webmachine.feature [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+          }
+        ]
+      }
+      """
+
+  Scenario: Delete resource
+    Given the item with uid "hello" exists
+    When the client does a DELETE request to the "item" resource with the template variable "item_uid" set to "hello"
+    Then the status code should be "204" (No Content)
+    And the item with uid "hello" should not exist
```
Comment (eterps) [inline features/webmachine.feature]: Nice set of examples.

File: lib/eipiai/webmachine/ext/decision.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      # response.
+      #
+      # Defaults to false
+      #
+      # @return [true, false] whether the request requires a payment
+      #
+      def payment_required?
+        false
+      end
+    end
```
Comment (eterps) [inline lib/eipiai/webmachine/ext/decision.rb]: Nice :-)

File: features/webmachine.feature (hunk 1 above)
Comment (JeanMertz) [inline features/webmachine.feature]: Yes :+1:, I'm planning on adding the [auto-curries feature](https://github.com/blendle/hal-representers/blob/master/lib/blendle/hal_representers/concerns/representable/json_ext.rb) that we have in hal-representers soon.

Comment (eterps) [inline features/webmachine.feature]: Ha ha, that would be a lovely feature.

Comment (JeanMertz) [inline features/webmachine.feature]: This has been implemented in https://github.com/blendle/eipiai/pull/6

**Pull request merged**